index.bs

```diff
@@ -575,68 +575,182 @@ urlPrefix: https://w3c.github.io/FileAPI/#dfn-; type: dfn;
+		Making the range of cells available as an image will allow the user to paste the cells into a photo editing app, while the text/plain format can be used by text editor apps.
+
+		A [=clipboard item=] can also optionally have a <dfn>presentation style</dfn> that helps distinguish whether apps "pasting" a [=clipboard item=] should insert the contents of an appropriate [=representation=] inline at the point of paste or if it should be treated as an attachment.
+
+		Apps that support pasting only a single [=clipboard item=] should use the first [=clipboard item=].
+		Apps that support pasting more than one [=clipboard item=] could, for example, provide a user interface that previews the contents of each [=clipboard item=] and allow the user to choose which one to paste.
+		Further, apps are expected to enumerate the [=mime type=]s of the [=clipboard item=] they are pasting and select the one best-suited for the app according to some app-specific algorithm.
+		Alternatively, an app can present the user with options on how to paste a [=clipboard item=], e.g. "paste as image" or "paste formatted text", etc.
+
+		A {{ClipboardItem}} object has an associated [=ClipboardItem/clipboard item=], which is a [=clipboard item=].
+
+		To create a {{ClipboardItem}} object, given a [=clipboard item=] |clipboardItem|'s [=relevant realm=] |realm|, run these steps:
+			1. Let |clipboardItemObject| be a new {{ClipboardItem}} with |realm|.
+
+			1. Set |clipboardItemObject|'s [=clipboard item=] to |clipboardItem|.
+
+		<a constructor lt="ClipboardItem()">constructor</a> steps for <code>new ClipboardItem(<var>items</var>, <var>options</var>)</code> are:
+			1. Set [=this=]'s [=ClipboardItem/clipboard item=] to a new [=clipboard item=].
+
+			1. If |options| is empty, then set [=this=]'s [=ClipboardItem/clipboard item=]'s [=presentation style=] to "unspecified", else, set [=this=]'s [=ClipboardItem/clipboard item=]'s [=presentation style=] to |options|["presentationStyle"].
+
+			1. For each (|key|, |value|) in |items|:
+
+				1. Let |mimeType| be the result of [=parse a mime type=] given |key|.
+
+				1. If |mimeType| is failure, then throw a {{TypeError}}.
+
+				1. Let |representation| be a new [=representation=].
+
+				1. Set |representation|'s [=mime type=] to |key|.
+
+				1. Set |representation|'s [=data=] to |value|.
+
+				1. Append |representation| to [=this=]'s [=ClipboardItem/clipboard item=]'s [=list of representations=].
+
+		<h4 attribute for=ClipboardItem lt=presentationStyle>presentationStyle</h4>
+		<p>
+			The {{ClipboardItem/presentationStyle}} getter steps are to return [=this=]'s [=ClipboardItem/clipboard item=]'s [=presentation style=].
+		</p>
+
+		<h4 attribute for=ClipboardItem lt=types>types</h4>
+		<p>
+			{{ClipboardItem/types}} getter steps are:
+
+			1. Let |types| be the result of running [=create a frozen array=].
```

Comment:
Unfortunately this version will create a new frozen array every time, since the getter steps are run every time you do `clipboardItem.types`. I.e., this will end up with `clipboardItem.types !== clipboardItem.types`.

Instead you should move this frozen array creation into the constructor, and store it on a new internal slot, e.g. something like `A {{ClipboardItem}} object has an associated <dfn for="ClipboardItem">types array</dfn>, which is a `FrozenArray<DOMString>`. Then this can just be `Return [=this=]'s [=ClipboardItem/types array=]`.

Reply:
Done.

Reply:
Ack.